src/bin/main.rs

```diff
@@ -0,0 +1,72 @@
+extern crate alloc;
+
+use alloc::sync::Arc;
+use clap::Parser;
+use nr_auth::commands::create::CreateCommand;
+use nr_auth::commands::retrieve_token::RetrieveTokenCommand;
+use nr_auth::http::client::HttpClient;
+use nr_auth::parameters::{
+    Commands, IdentityType, OutPutTokenFormat, build_token_for_identity_creation,
+    create_metadata_for_identity_creation, create_metadata_for_token_retrieve,
+};
+use std::boxed::Box;
+use std::error::Error;
+use std::fmt::Debug;
+use std::result::Result;
+use std::result::Result::Ok;
+
+#[derive(Parser, Debug)]
+#[command(name = "newrelic-auth-cli")]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+fn main() -> Result<(), Box<dyn Error>> {
+    let cli_command = Cli::parse();
+
+    let http_client = HttpClient::new()?;
+    let retrieve_token_command = RetrieveTokenCommand::new(Arc::new(http_client.clone()));
+    let create_command = CreateCommand::new(Arc::new(http_client.clone()));
```

Comment:
Would it make sense to create the `retrieve_token_command` and `create_command` variables inside the match (depending on the selected option).